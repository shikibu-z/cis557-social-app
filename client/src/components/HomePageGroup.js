import React from 'react';
import '../stylesheets/HomePageGroup.css';
import {
  Link,
} from 'react-router-dom';

function HomePageGroup(props) {
  const {
    id, src, groupName, groupInfo, members, priv, join,
  } = props;

  let isPrivate = 'No';
  let isJoined = 'No';
  if (priv === 1) {
    isPrivate = 'Yes';
  }
  if (join === 'true') {
    isJoined = 'Yes';
  }

  return (
    <Link key={id} style={{ textDecoration: 'none', color: 'black' }} to={`/explore/${id}`}>
      <div className="homepage-group">
        <img src={src} alt="" />
        <div className="homepage-group-name">
          <h2>{groupName}</h2>
          <p className="homepage-group-info">
            {groupInfo}
          </p>
          <p className="homepage-group-members">
            Private: {isPrivate}
            {' '}
            Joined: {isJoined}
            {' '}
            Members: {members}
          </p>
        </div>
      </div>
    </Link>
  );
}

export default HomePageGroup;
