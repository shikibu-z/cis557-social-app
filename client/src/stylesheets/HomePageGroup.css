.homepage-group {
  display: flex;
  flex-direction: row;
  align-self: center;
  background-color: white;
  width: 90%;
  height: auto;
  border-radius: 20px;
  box-shadow: 0px 5px 5px grey;
  margin: 0% auto 2%;
}

.homepage-group img {
  width:80px;
  height:80px;
  border-radius: 50%;
  overflow:hidden;
  margin: 2% 4% 1% 4%;
}

.homepage-group-name {
  flex:1;
  margin: 1% 5% 1% 1%;
  display: flex;
  flex-direction: column;
}

.homepage-group-name h2 {
  margin-top: 1%;
  font-size: 30px;
  margin-bottom: 2%;
}

.homepage-group-info{
  color:#595959;
  font-weight: 400;
  font-size: 25px;
  line-height: 24px;
}

.homepage-group-members {
  color: #8C8C8C;
  text-align: right;
  margin-right: 2%;
}