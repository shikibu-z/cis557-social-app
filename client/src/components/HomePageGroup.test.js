/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import HomePageGroup from './HomePageGroup';

test('HomePageGroup matches snapshot', () => {
  const component = renderer.create(<BrowserRouter><HomePageGroup groupName="investing" /></BrowserRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
