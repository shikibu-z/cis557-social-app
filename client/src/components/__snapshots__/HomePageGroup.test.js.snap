// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageGroup matches snapshot 1`] = `
<a
  href="/explore/undefined"
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
>
  <div
    className="homepage-group"
  >
    <img
      alt=""
    />
    <div
      className="homepage-group-name"
    >
      <h2>
        investing
      </h2>
      <p
        className="homepage-group-info"
      />
      <p
        className="homepage-group-members"
      >
        Private: 
        No
         
        Joined: 
        No
         
        Members: 
      </p>
    </div>
  </div>
</a>
`;
